bluetooth_hid_notes.md

```diff
@@ -104,6 +104,31 @@ MCU update state report?
 
 Standard full mode - input reports with IMU data instead of subcommand replies. Pushes current state @60Hz, or @120Hz if Pro Controller.
 
+If 6-axis sensor is enabled, the IMU data in an 0x30 input report starts at bit 13 and is packaged like this:
+
+```
+
+| Byte       | Remarks                                                                                          |
+|:----------:| ------------------------------------------------------------------------------------------------ |
+|   13-14    | accel_x (Int16LE); x-axis is parallel to the Joy-Con rail                                        |
```

Comment:
Please update bytes (they should start from 12).

And change the axis description to reflect this:
x-axis is vertical to the Joy-Con rail (+ is to the right)
y-axis is parallel to the Joy-Con rail (+ is forward)
z-axis is parallel to the joystick (+ is up)

Reply:
I don't think I agree with you on the axis descriptions. When I set the Joy-Con down so the rail is on the table, my second value goes to 1, so that would indicate to me that the y-axis is parallel to the line between the Y and A buttons. When I stand it on the IR sensor the first value goes to 1, so I think that means the x-axis is parallel to the Joy-Con rail (or the line between the X and B buttons). That leaves the z-axis to be parallel to the joystick. Do you get results that don't agree with this? I've only tested this on a right Joy-Con.

Reply:
So to better understand, because I can only see the raw values right now,

When a +Axis looks earth, that value must be the biggest.
And if on -Axis, the smallest.
If the axis is parallel to earth it's near 0 because no force is applied along its axis.

That much is true (correct me if the -+ is the opposite). 

But how you define which is X/Y/Z if not on how you hold it?
For example, stick's default XY is when holding the joy-con normally.
If this is the case, shouldn't be YXZ for acc?

With XYZ, the values match the sideways, movement.

Anyway, what I want to address is the following: 
because we assume that we hold the joy-con normally, you need to clarify, if this changes.
So, maybe add (Joy-Con sideways).

----

Btw, it seems that with joy-con sideways (and assuming I got the -+ above right), the axis are like this:
Left:
X -> Normal
Y -> Normal
Z -> Reverse

Right:
X -> Reverse
Y -> Normal
Z -> Normal

Reply:
I'm defining x/y/z relative to the accelerometer, based on what value is passed first in the input report. It's independent of how the joy-con is held. X is passed first, followed by Y, then Z. I didn't state which way is positive and which is negative in my edits; I only tested on one joy-con so far. I just went to measure them now.

Without scaling I think the range of the accel data should be +/- 4096. So at a glance, if you see something close to 0x0fff or 0xf001 (big endian) in an input report you know the axis is roughly parallel to the ground. Here's a shitty ascii art of the data I get lol. If you agree with these results I can definitely pack it into the pull request so it's a little clearer.
I'm defining it like this: Positive [axis] means that when the Joy-Con is placed with this side up, you read the largest [axis] value from the accelerometer, and vice versa. That sign convention is arbitrary; reversing all the signs would be equally valid; it's just worth being consistent.
```
_______
|      \
|       |        +x +z
|   R   |    -y __|/__ +y
|       |        /|
|       |     -z -x
|______/

```
I'm seeing that the axis I'm calling Z (the axis parallel to the stick of the joystick) is reversed. (Have only tested with right Joy-Con still.) This actually makes sense -the accel reports are relative to the orientation of the IMU chip. [The second image on this page](https://github.com/dekuNukem/Nintendo_Switch_Reverse_Engineering/blob/master/joycon_pcb.pdf) shows that the IMU is mounted "upside down" on the bottom of the Joy-Con PCB, so it makes sense that its Z axis would be flipped.

Reply:
Thank you for your write up. Everything is clear now.
If you want to include that info, be my guest. The more info the better.

Make sure to also add the reverse info in your bytes table also.
```
accel_x (Int16LE); x-axis is parallel to the Joy-Con rail (reversed in Left Joy-Con)
--
accel_z (Int16LE); z-axis is parallel to the joystick  (reversed in Right Joy-Con)
```